vm/qemu/qmp.go

```diff
@@ -53,29 +48,42 @@ func (inst *instance) qmpConnCheck() error {
 		return err
 	}
 
-	monDec := json.NewDecoder(conn)
 	monEnc := json.NewEncoder(conn)
 
-	var banner qmpBanner
-	if err := monDec.Decode(&banner); err != nil {
-		return err
-	}
+	inst.mon = conn
+	inst.qmpRecv()
 
 	inst.monEnc = monEnc
-	inst.monDec = monDec
 	if _, err := inst.doQmp(&qmpCommand{Execute: "qmp_capabilities"}); err != nil {
 		inst.monEnc = nil
-		inst.monDec = nil
+		inst.mon = nil
 		return err
 	}
-	inst.mon = conn
 
 	return nil
 }
 
 func (inst *instance) qmpRecv() (*qmpResponse, error) {
 	qmp := new(qmpResponse)
-	err := inst.monDec.Decode(qmp)
+	var err error
+
+	scanner := bufio.NewScanner(inst.mon)
```

Comment:
bufio.Scanner is buffered reader, we should not create it each time and discard. This may discard some unrelated output.